### PR TITLE
transaction date: set now as default in loader

### LINKED ISSUE #294: Loan start date
Package version (if known): alpha.17

## Describe the bug
The start date assigned by the system is not per expectations, but some date in the past that doesn't seem correlated to the user request.

## Steps to Reproduce
Checked out an item for a loan request, but the actual start date (and end date) was not updated to the check out date (today), per normal expectations.  Instead it is assigning some date in the past, which is neither the request date ('period of interest starts' date) or today (day of check out)

## Expected behavior
Maybe the date is set per some internal criteria that I may not understand, but doesn't seem right per common library procedures and also creates problems - in some cases, the actual date is so past the request date, that it shows OVERDUE immediately upon check out itself.

## Screenshots (if applicable)
First shot shows Pending Loan request for which I checked out PID -v3p40
Second shot shows Start/End date assigned which is not the Check out date (today) or the Requested Start date.
Further, it is immediately/already Overdue!

Screen Shot 2020-11-07 at 2 36 29 PM
Screen Shot 2020-11-07 at 2 38 26 PM

## Additional context

### PR DESCRIPTION
**Pull request by KonstantinaStoikou**
Closes inveniosoftware/react-invenio-app-ils#294

### CODE REVIEW

File: invenio_circulation/records/loaders/schemas/json.py [hunk 1]
```diff
@@ -92,7 +95,7 @@ def get_pid_field(self):
     request_expire_date = DateString()
     request_start_date = DateString()
     start_date = DateString()
-    transaction_date = DateTimeString()
+    transaction_date = DateTimeString(missing=transaction_date_default)
```
Comment (kpsherva) [inline invenio_circulation/records/loaders/schemas/json.py]:
```suggestion
    transaction_date = DateTimeString(missing=arrow.utcnow())
```
can be a value or callable

File: invenio_circulation/transitions/base.py [hunk 2]
```diff
@@ -172,7 +172,6 @@ def _date_fields2datetime(self, kwargs):
     def before(self, loan, initial_loan, **kwargs):
         """Validate input, evaluate conditions and raise if failed."""
         loan.update(kwargs)
-        loan.setdefault("transaction_date", arrow.utcnow())
```
Comment (kpsherva) [inline invenio_circulation/transitions/base.py]: I would leave it in case the transaction is triggered programmaticaly, not via REST, ping @ntarocco

**Review by jrcastro2 — APPROVED** [review]

File: invenio_circulation/records/loaders/schemas/json.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def get_pid_field(self):
     request_expire_date = DateString()
     request_start_date = DateString()
     start_date = DateString()
-    transaction_date = DateTimeString()
+    transaction_date = DateTimeString(missing=arrow.get().utcnow().isoformat())
```
Comment (ntarocco) [inline invenio_circulation/records/loaders/schemas/json.py]:
```suggestion
    transaction_date = DateTimeString(missing=arrow.utcnow().isoformat())
```

**Review by ntarocco — APPROVED** [review]

**Review by ntarocco — APPROVED** [review]

**Comment by ntarocco** [comment]
> @BadrAly is this ok? Can we merge?
> One Travis job failed, but not related to the code and we cannot easily restart it.

@ntarocco tested and all is OK

**Review by BadrAly — APPROVED** [review]

**Pull request merged**